utils_nlp/models/transformers/sequence_classification.py

```diff
@@ -1,2 +1,147 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
-# Licensed under the MIT License.
\ No newline at end of file
+# Licensed under the MIT License.
+
+import torch
+from torch.utils.data import TensorDataset
+
+from pytorch_transformers.modeling_bert import (
+    BERT_PRETRAINED_MODEL_ARCHIVE_MAP,
+    BertForSequenceClassification,
+)
+from pytorch_transformers.modeling_distilbert import (
+    DISTILBERT_PRETRAINED_MODEL_ARCHIVE_MAP,
+    DistilBertForSequenceClassification,
+)
+from pytorch_transformers.modeling_roberta import (
+    ROBERTA_PRETRAINED_MODEL_ARCHIVE_MAP,
+    RobertaForSequenceClassification,
+)
+from pytorch_transformers.modeling_xlnet import (
+    XLNET_PRETRAINED_MODEL_ARCHIVE_MAP,
+    XLNetForSequenceClassification,
+)
+
+from utils_nlp.models.transformers.common import MAX_SEQ_LEN, TOKENIZER_CLASS, fine_tune
+
+MODEL_CLASS = {}
+MODEL_CLASS.update({k: BertForSequenceClassification for k in BERT_PRETRAINED_MODEL_ARCHIVE_MAP})
+MODEL_CLASS.update(
+    {k: RobertaForSequenceClassification for k in ROBERTA_PRETRAINED_MODEL_ARCHIVE_MAP}
+)
+MODEL_CLASS.update({k: XLNetForSequenceClassification for k in XLNET_PRETRAINED_MODEL_ARCHIVE_MAP})
+MODEL_CLASS.update(
+    {k: DistilBertForSequenceClassification for k in DISTILBERT_PRETRAINED_MODEL_ARCHIVE_MAP}
+)
+
+
+def _list_supported_models():
+    return list(MODEL_CLASS)
+
+
+class Processor:
+    def __init__(self, model_name, tokenize=None, to_lower=False, cache_dir="."):
+        self.tokenizer = TOKENIZER_CLASS[model_name].from_pretrained(
+            model_name, do_lower_case=to_lower, cache_dir=cache_dir
+        )
+        self.custom_tokenize = tokenize
+
+    @staticmethod
+    def get_inputs(batch, model_name):
+        if model_name.split("-")[0] == "bert":
+            return {"input_ids": batch[0], "attention_mask": batch[1], "labels": batch[2]}
+        else:
+            raise ValueError("Model not supported.")
+
+    @staticmethod
+    def list_supported_models():
```

Comment:
@saidbleik why do we need this as a wrapper of a wrapper if we end up having it as a static method.